testing/MLDB-2188-label_feature_validation.py

```diff
@@ -0,0 +1,216 @@
+                        "limit" : 10000,
+                    }
+                }) 
+
+        mldb.put("/v1/procedures/baggify", {
+            "type": "transform",
+            "params": {
+                "inputData": """
+                select Theme, tokenize(lower(Body), {splitChars:' ,.:;«»[]()%!?', quoteChar:'', minTokenLength: 2}) 
+                as tokens from src_train       
+                """,
+                "outputDataset": {
+                    "id": "bag_of_words",
+                    "type": "sparse.mutable"
+                },
+            }
+        })
+
+    def test_categorical(self):
+
+        cls_config = {
+                "my_fasttext": {
+                    "type": "fasttext",
+                    "verbosity" : 0,
+                    "dims" : 4,
+                    "epoch" : 5,
+                }
+            }
+
+        with self.assertMldbRaises(expected_regexp="Dataset column 'Theme' is used in both label and feature"):
+            mldb.put("/v1/procedures/trainer", {
+                "type": "classifier.train",
+                "params": {
+                    "trainingData": "SELECT {Theme} as features, Theme as label FROM bag_of_words",
+                    "modelFileUrl": "file://tmp/src_fasttext.cls",
+                    "functionName" : 'myclassify',
+                    "algorithm": "my_fasttext",
+                    "mode": "categorical",
+                    "runOnCreation": True,
+                    "configuration": cls_config
+                }
+            })
+
+    def test_categorical_expression(self):
+
+        cls_config = {
+                "my_fasttext": {
+                    "type": "fasttext",
+                    "verbosity" : 0,
+                    "dims" : 4,
+                    "epoch" : 5,
+                }
+            }
+
+        with self.assertMldbRaises(expected_regexp="classifier.train only accepts wildcard and column names"):
+            mldb.put("/v1/procedures/trainer", {
+                "type": "classifier.train",
+                "params": {
+                    "trainingData": "SELECT {Theme} as features, Theme + '_catsup' as label FROM bag_of_words",
+                    "modelFileUrl": "file://tmp/src_fasttext.cls",
+                    "functionName" : 'myclassify',
+                    "algorithm": "my_fasttext",
+                    "mode": "categorical",
+                    "runOnCreation": True,
+                    "configuration": cls_config
+                }
+            })
+    
+    def test_wildcard(self):
+
+        cls_config = {
+                "my_fasttext": {
+                    "type": "fasttext",
+                    "verbosity" : 0,
+                    "dims" : 4,
+                    "epoch" : 5,
+                }
+            }
+
+        with self.assertMldbRaises(expected_regexp="Dataset column 'tokens.alabama' is used in both label and feature because of label wildcard 'tokens'"):
```

Comment:
Making multiline strings is a pain though because then you can't use multiline strings to comment out some tests.